bot.py

```diff
@@ -222,8 +229,13 @@ def check(m):
 	await channel.send(file=file, embed=embed)
 
 	await pokemon.save()
-	profile = userprofile.UserProfile()
-	profile.user_id = message.author.id
+
+	if profile is None:
+		await channel.send(
+			f"{message.author.mention} error while saving. Profile not found"
+		)
+		return
```

Comment:
This check should be done in the while loop, and it shouldn't kill the game. It should also tell the user to create a profile with `p!begin` instead.